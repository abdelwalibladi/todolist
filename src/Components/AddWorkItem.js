import React from "react";
import {
  Text,
  View,
  TextInput,
  TouchableOpacity,
  Dimensions,
  StyleSheet
} from "react-native";

const addWorkItem = props => {
  return (
    <View style={styles.container}>
      <View style={styles.popUpBox}>
        <View style={styles.popUpBoxItem}>
          <Text style={styles.action}>title:</Text>
          <TextInput
            style={styles.input}
            placeholder="ex : Conférence Mercredi à 15h"
            underlineColorAndroid="black"
            onChange={props.inserWorkItem}
            placeholderTextColor="white"
          />
          <Text style={styles.action}>Détail:</Text>
          <TextInput
            style={styles.input}
            underlineColorAndroid="black"
            onChange={props.inserDetail}
            placeholder="ex :Aws Submit 2020"
            placeholderTextColor="white"
          />
        </View>
        <View style={styles.actionBox}>
          <TouchableOpacity onPress={props.cancel}>
            <Text style={styles.cancel}>Retour</Text>
          </TouchableOpacity>
          <TouchableOpacity onPress={props.save}>
            <Text style={styles.save}>Enregister</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: "100%",
    alignItems: "center",
    height: Dimensions.get("screen").height,
    flex: 2,
    marginBottom: 100
  },
  popUpBox: {
    backgroundColor: "#ccc",
    width: "95%",
    top: "5%",
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10
  },
  popUpBoxItem: {
    height: 140,
    width: "100%",
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "column"
  },
  input: {
    width: "80%",
    left: 10,
    color: "black",
    fontSize: 15,
    height: 40
  },
  actionBox: {
    justifyContent: "space-around",
    flexDirection: "row",
    marginBottom: 20
  },
  cancel: {
    color: "red",
    fontSize: 20,
    borderColor: "white"
  },
  save: {
    color: "#fff",
    fontSize: 20
  }
});
export default addWorkItem;
