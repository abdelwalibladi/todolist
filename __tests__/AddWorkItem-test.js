import React from "react";
import AddWorkItem from "../src/Components/AddWorkItem";
import renderer from "react-test-renderer";

test("addwork item snapshot", () => {
  const snap = renderer.create(<AddWorkItem />).toJSON();

  expect(snap).toMatchSnapshot();
});

it("render correctly", () => {
  const snap = renderer.create(<AddWorkItem />);
});
