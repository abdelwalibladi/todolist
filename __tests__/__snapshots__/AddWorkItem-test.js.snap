// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addwork item snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 2,
      "height": 1334,
      "marginBottom": 100,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#555555",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "top": "5%",
        "width": "95%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": 140,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text>
        Titre:
      </Text>
      <TextInput
        allowFontScaling={true}
        placeholder="  ex : Conférence Mercredi à 15h"
        placeholderTextColor="white"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 0.5,
            "color": "white",
            "fontSize": 15,
            "height": 40,
            "left": 10,
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text>
        Détail:
      </Text>
      <TextInput
        allowFontScaling={true}
        placeholder="  ex : Aws Submit 2020"
        placeholderTextColor="white"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 0.5,
            "color": "white",
            "fontSize": 15,
            "height": 40,
            "left": 10,
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginBottom": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "white",
              "color": "red",
              "fontSize": 20,
            }
          }
        >
          Retour
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
            }
          }
        >
          Enregister
        </Text>
      </View>
    </View>
  </View>
</View>
`;
